### ISSUE TITLE
Null Object Reference crash when Azra dies (different stack trace)

### ISSUE THREAD

**Issue by IBwWG** [larsiusprime/tdrpg-bugs#908]
0.8.4, win7x64, HD mode@1704x960windowed

Different trace than #719.

![dq-crash-azra-dies2](https://cloud.githubusercontent.com/assets/16719964/15806582/16fbd7f8-2b46-11e6-98d1-ac953355d84e.png)

You can use the save file from #907 .  Going back in, it again didn't save my battle layout with the crash.

Although, doesn't seem related to layout.  Tried a completely different one (also ineffective)--she dies, click OK, animations play a second or two, then crash instead of fading out.

```
Called from ApplicationMain::main ApplicationMain.hx line 159
Called from ApplicationMain::create ApplicationMain.hx line 71
Called from lime.app.Application::exec lime/app/Application.hx line 222
Called from lime._backend.native.NativeApplication::exec lime/_backend/native/Na
tiveApplication.hx line 136
Called from lime._backend.native.NativeApplication::handleRenderEvent lime/_back
end/native/NativeApplication.hx line 368
Called from lime.app.Event_Void_Void::dispatch lime/app/Event.hx line 145
Called from *::_Function_2_1 lime/app/Application.hx line 110
Called from lime.app.Application::render lime/app/Application.hx line 762
Called from openfl.display.Stage::render openfl/display/Stage.hx line 761
Called from openfl.display.DisplayObjectContainer::__broadcast openfl/display/Di
splayObjectContainer.hx line 355
Called from openfl.display.DisplayObject::__broadcast openfl/display/DisplayObje
ct.hx line 251
Called from openfl.events.EventDispatcher::__dispatchEvent openfl/events/EventDi
spatcher.hx line 237
Called from flixel.FlxGame::onEnterFrame flixel/FlxGame.hx line 530
Called from flixel.FlxGame::step flixel/FlxGame.hx line 663
Called from com.leveluplabs.tdrpg.Main::update com/leveluplabs/tdrpg/Main.hx lin
e 1495
Called from flixel.FlxGame::update flixel/FlxGame.hx line 752
Called from flixel.FlxState::tryUpdate flixel/FlxState.hx line 168
Called from com.leveluplabs.tdrpg.State_Battle::update com/leveluplabs/tdrpg/Sta
te_Battle.hx line 905
Called from com.leveluplabs.tdrpg.State_Battle::battleUpdate com/leveluplabs/tdr
pg/State_Battle.hx line 1020
Called from com.leveluplabs.tdrpg.BattleModel::update com/leveluplabs/tdrpg/Batt
leModel.hx line 1421
Called from com.leveluplabs.tdrpg.BattleModel::doUpdate com/leveluplabs/tdrpg/Ba
ttleModel.hx line 1831
Called from com.leveluplabs.tdrpg.BattleModel::updateCollisions com/leveluplabs/
tdrpg/BattleModel.hx line 3129
Called from com.leveluplabs.tdrpg.BattleModel::updateProjectileCollisions com/le
veluplabs/tdrpg/BattleModel.hx line 3180
Called from com.leveluplabs.tdrpg.BattleModel::projectileCollision com/levelupla
bs/tdrpg/BattleModel.hx line 2939
Called from com.leveluplabs.tdrpg.CreatureSprite::hitByProjectile com/leveluplab
s/tdrpg/CreatureSprite.hx line 2055
Called from com.leveluplabs.tdrpg.CreatureSprite::resolveAttack com/leveluplabs/
tdrpg/CreatureSprite.hx line 2075
Called from com.leveluplabs.tdrpg.EnemySprite::takeDamage com/leveluplabs/tdrpg/
EnemySprite.hx line 839
Called from com.leveluplabs.tdrpg.CreatureSprite::takeDamageFlavors com/levelupl
abs/tdrpg/CreatureSprite.hx line 1551
Called from com.leveluplabs.tdrpg.EnemySprite::applyFlavor com/leveluplabs/tdrpg
/EnemySprite.hx line 1028
Called from com.leveluplabs.tdrpg.CreatureSprite::applyFlavor com/leveluplabs/td
rpg/CreatureSprite.hx line 486
Called from com.leveluplabs.tdrpg.CreatureSprite::applyStatus com/leveluplabs/td
rpg/CreatureSprite.hx line 445
Called from flixel.util._FlxSignal.FlxSignal2::dispatch2 flixel/util/FlxSignal.h
x line 292
Called from com.leveluplabs.tdrpg.State_Battle::onSignalUI com/leveluplabs/tdrpg
/State_Battle.hx line 2169
Called from com.leveluplabs.tdrpg.State_Battle::onSelectEnemy com/leveluplabs/td
rpg/State_Battle.hx line 1970
Called from com.leveluplabs.tdrpg.State_Battle::showBattleEnemyMenu com/levelupl
abs/tdrpg/State_Battle.hx line 4294
Called from com.leveluplabs.tdrpg.UI_BattleEnemyMenu::show com/leveluplabs/tdrpg
/UI_BattleEnemyMenu.hx line 281
Called from com.leveluplabs.tdrpg.UI_BattleEnemyMenu::showEnemyStats com/levelup
labs/tdrpg/UI_BattleEnemyMenu.hx line 454
```

**Comment by ManDude** [comment]
Just wondering, how are you getting these stack traces?

**Comment by IBwWG** [comment]
Just start the game in a command prompt window; when there's a crash, they get spit out there.

**Comment by larsiusprime** [comment]
Fixed in 0.8.9 -- should already be fixed by the better battle end resolution in 0.8.8, but I'm adding extra guards for this too
